pkg/astnormalization/extends_directive.go

```diff
@@ -0,0 +1,56 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+/*
+@extends directive on an object will make it a type extension
+type User {
+   id: ID!
+   name: String
+}
+     +
+type User @key(fields: "id") @extends {
+   id: ID! @external
+   name: String
+}
+
+will be,
+
+type User {
+   id: ID!
+   name: String
+}
+*/
+type extendsDirectiveVisitor struct {
+	operation *ast.Document
```

Comment:
`operation` may not be the right term here as we don't have an operation. 
`document` will be enough

Reply:
Also, can we have some tests for this visitor?

Reply:
there is a tests for this